Spigot-API-Patches/0284-Reusable-ChunkSnapshot-API.patch

```diff
@@ -0,0 +1,34 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Paul Sauve <paul@technove.co>
+Date: Wed, 31 Mar 2021 09:19:58 -0500
+Subject: [PATCH] Reusable ChunkSnapshot API
+
+
+diff --git a/src/main/java/org/bukkit/Chunk.java b/src/main/java/org/bukkit/Chunk.java
+index 98263d896f316983609432c45b85401a2692432d..7b8710d0e720d1209d31f51b546e1df1b3104cae 100644
+--- a/src/main/java/org/bukkit/Chunk.java
++++ b/src/main/java/org/bukkit/Chunk.java
+@@ -97,6 +97,23 @@ public interface Chunk extends PersistentDataHolder {
+     @NotNull
+     ChunkSnapshot getChunkSnapshot(boolean includeMaxblocky, boolean includeBiome, boolean includeBiomeTempRain);
+ 
++    // Paper start
++    /**
++     * Capture thread-safe read-only snapshot of chunk data, reusing previous snapshot
++     *
++     * @param snapshot - previous snapshot to reuse
++     * @param includeMaxblocky - if true, snapshot includes per-coordinate
++     *     maximum Y values
++     * @param includeBiome - if true, snapshot includes per-coordinate biome
++     *     type
++     * @param includeBiomeTempRain - if true, snapshot includes per-coordinate
++     *     raw biome temperature and rainfall
++     * @return ChunkSnapshot
++     */
++    @NotNull
++    ChunkSnapshot getChunkSnapshot(@org.jetbrains.annotations.Nullable ChunkSnapshot snapshot, boolean includeMaxblocky, boolean includeBiome, boolean includeBiomeTempRain);
```

Comment:
`includeMaxblocky` -> `includeMaxBlockY` follows camelCase a bit better in my opinion.

Reply:
Probably, but the method without reusing the chunksnapshot has the same name for this parameter. I don't think it's the job of this patch to redo the naming convention for the parameter.

Reply:
Oh, I see the interface defines the parameter `includeMaxblocky` while the implementation defines it `includeMaxBlockY` (in the case of the method you are trying to overload).

Seems like a simple error on the interface definition of the existing method, I guess your choice if you want to be consistent with that error or not.